plugins/plugin-chart-echarts/src/Graph/controlPanel.tsx

```diff
@@ -0,0 +1,270 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import { t } from '@superset-ui/core';
+import {
+  sections,
+  sharedControls,
+  ColumnOption,
+  ColumnMeta,
+} from '@superset-ui/chart-controls';
+import { DEFAULT_FORM_DATA } from './types';
+import {
+  legendMarginControl,
+  legendOrientationControl,
+  legendTypeControl,
+  showLegendControl,
+} from '../controls';
+
+const noopControl = { name: 'noop', config: { type: '', renderTrigger: true } };
+const columnSelectControl = {
+  ...sharedControls.entity,
+  type: 'SelectControl',
+  multi: false,
+  freeForm: true,
+  default: null,
+  includeTime: false,
+  optionRenderer: (c: ColumnMeta) => <ColumnOption column={c} showType />,
+  valueRenderer: (c: ColumnMeta) => <ColumnOption column={c} />,
+  valueKey: 'column_name',
+  allowAll: true,
+  filterOption: ({ data: opt }: { data: ColumnMeta }, text: string) =>
+    (opt.column_name && opt.column_name.toLowerCase().indexOf(text.toLowerCase()) >= 0) ||
+    (opt.verbose_name && opt.verbose_name.toLowerCase().indexOf(text.toLowerCase()) >= 0),
+  promptTextCreator: (label: string) => label,
+  commaChoosesOption: false,
+};
+
+const sourceControl = {
+  name: 'source',
+  config: {
+    ...columnSelectControl,
+    clearable: false,
+    label: t('Source'),
+    description: t('Name of the source nodes'),
+  },
+};
+
+const targetControl = {
+  name: 'target',
+  config: {
+    ...columnSelectControl,
+    clearable: false,
+    label: t('Target'),
+    description: t('Name of the target nodes'),
+  },
+};
+
+const categoryControl = {
+  name: 'category',
+  config: {
+    ...columnSelectControl,
+    clearable: true,
+    label: t('Color By'),
+    description: t('Optional category for nodes of graph'),
+    validators: [],
+  },
+};
+
+export default {
+  controlPanelSections: [
+    sections.legacyRegularTime,
+    {
+      label: t('Query'),
+      expanded: true,
+      controlSetRows: [
+        [sourceControl],
+        [targetControl],
+        ['metric'],
+        [categoryControl],
+        ['adhoc_filters'],
+        ['row_limit'],
+      ],
+    },
+    {
+      label: t('Chart Options'),
+      expanded: true,
+      controlSetRows: [
+        ['color_scheme'],
+        [<h1 className="section-header">{t('Legend')}</h1>],
+        [showLegendControl],
+        [legendTypeControl, legendOrientationControl],
+        [legendMarginControl, noopControl],
+        [<h1 className="section-header">{t('Layout')}</h1>],
+        [
+          {
+            name: 'layout',
+            config: {
+              type: 'RadioButtonControl',
+              renderTrigger: true,
+              label: t('Graph Layout'),
+              default: DEFAULT_FORM_DATA.layout,
+              options: [
+                {
+                  label: 'force',
```

Comment:
How do I use t() here, to make it available for translation?

Reply:
I think it currently works? As long as you don't import `controlPanel` synchronously in a chart plugin's `index.ts` (which will be called before `configureTranslation()` is called), you should be fine.